src/modelfree/configs/multi/train.py

```diff
@@ -15,6 +17,90 @@
 TARGET_VICTIM = collections.defaultdict(lambda: 1)
 TARGET_VICTIM['multicomp/KickAndDefend-v0'] = 2
 
+HYPERPARAM_SEARCH_VALUES = {
+    'seed': tune.sample_from(
+        lambda spec: np.random.randint(1000)),
+
+    # Dec 2018 experiments used 2^11 = 2048 batch size.
+    # Aurick Zhou used 2^14 = 16384; Bansal et al use 409600 ~= 2^19.
+    'batch_size': tune.sample_from(
+        lambda spec: 2 ** np.random.randint(11, 16)),
+
+    'rl_args': {
+        # PPO2 default is 0.01. run_humanoid.py uses 0.00.
+        'ent_coef': tune.sample_from(
+            lambda spec: np.random.uniform(low=0.00, high=0.02)),
+
+        # nminibatches must be a factor of batch size; OK provided power of two
+        # PPO2 default is 2^2 = 4; run_humanoid.py is 2^5 = 32
+        'nminibatches': tune.sample_from(
+            lambda spec: 2 ** (np.random.randint(0, 7))),
+
+        # PPO2 default is 4; run_humanoid.py is 10
+        'noptepochs': tune.sample_from(
+            lambda spec: np.random.randint(1, 11)),
+                },
+    # PPO2 default is 3e-4; run_humanoid uses 1e-4;
+    # Bansal et al use 1e-2 (but with huge batch size).
+    # Sample log-uniform between 1e-2 and 1e-5.
+    'learning_rate': tune.sample_from(
+        lambda spec: 10 ** (-2 + -3 * np.random.random())),
+}
+
+MULTI_TRAIN_LOCATION = osp.join(os.environ.get('DATA_LOC', 'data'), "multi_train")
+
+# TODO Will move this into a JSON at some point, keeping here for easier debugging for the moment
```

Comment:
Okay. It's always a bit painful to see the dependency chain of different bits of code that need to be run to do an experiment get longer, since you then have to have outside oral-tradition knowledge about the fact that this Sacred experiment can't actually be run by itself unless you've already run highest_win_rate first. But that is how we do it in the rest of the codebase, and it's probably better that it be consistent.

Reply:
I'm not sure I understand this objection.

Hard-coding the paths seems maximally fragile. If someone doesn't have all of our data, they can't run this experiment.

Whereas with loading things from a JSON, yes they have to run this extra command, but we can either document this or write a script which does an entire pipeline (trains adversaries, evaluates best ones, finetunes, etc).

I think we do save the JSON to S3, so if someone is copying all of our data, I think both are similarly easy.